awx/main/models/mixins.py

```diff
@@ -464,15 +464,24 @@ class Meta:
 
     def resolve_execution_environment(self):
         """
-        Return the execution environment that should be used when creating a new job.
+        Return the execution environment that should be used when executing a job.
         """
         if self.execution_environment is not None:
             return self.execution_environment
+        template = getattr(self, 'unified_job_template', None)
+        if template is not None and template.execution_environment is not None:
+            return template.execution_environment
+        wf_node = getattr(self, 'unified_job_node', None)
+        while wf_node is not None:
+            wf_template = wf_node.workflow_job.workflow_job_template
```

Comment:
Can't quite articulate when/why but feel like `workflow_job` and `workflow_job_template` _could_ be null.

Reply:
I believe that's further addressed in #10351

Reply:
(also, it's part of the already merged PR)